artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java

```diff
@@ -1284,7 +1284,7 @@ private boolean checkDuplicateID(final Message message,
          boolean isDuplicate = false;
 
          if (duplicateIDBytes != null) {
-            cache = getDuplicateIDCache(message.getAddressSimpleString());
+            cache = getDuplicateIDCache(context.getAddress(message));
```

Comment:
Same as other comment, re making one call to context.getAddress(message) and re-using the result

Reply:
@michaelandrepearce caching here goes beyond the scope here.

I would need to change a lot of signatures to pass an address as a parameter.. or cache within each method used during routing.


We can do that.. but it goes beyond the scope of this fix.

Reply:
No worries I can do separate pr later